lisa/transformers/kernel_installer.py

```diff
@@ -0,0 +1,308 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+
+import re
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Optional, Type, cast
+
+from dataclasses_json import CatchAll, Undefined, dataclass_json
+
+from lisa import schema
+from lisa.node import Node, RemoteNode
+from lisa.operating_system import Ubuntu
+from lisa.secret import PATTERN_HEADTAIL, add_secret
+from lisa.tools import Cat, Uname
+from lisa.transformer import Transformer
+from lisa.util import get_matched_str, subclasses
+from lisa.util.logger import Logger, get_logger
+
+
+@dataclass_json(undefined=Undefined.INCLUDE)
+@dataclass
+class BaseInstallerSchema(schema.TypedSchema):
+    delay_parsed: CatchAll = field(default_factory=dict)  # type: ignore
```

Comment:
Why have this base? The field is being redeclared (it has to, given it's a pre-populated value one).

Reply:
Removed the redeclared fields in its subclasses. The base schema is used to parse type in factory, and save other fields in the delay_parsed.